pkg/decision/forced_decision_service_test.go

```diff
@@ -0,0 +1,213 @@
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package decision //
+package decision
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/optimizely/go-sdk/pkg/config"
+	"github.com/optimizely/go-sdk/pkg/decide"
+	"github.com/stretchr/testify/suite"
+)
+
+var doOnce sync.Once // required since we only need to read datafile once
+var datafile []byte
+
+type ForcedDecisionServiceTestSuite struct {
+	suite.Suite
+	forcedDecisionService *ForcedDecisionService
+	projectConfig         config.ProjectConfig
+}
+
+func (s *ForcedDecisionServiceTestSuite) SetupTest() {
+	s.forcedDecisionService = NewForcedDecisionService("abc")
+	doOnce.Do(func() {
+		absPath, _ := filepath.Abs("../../test-data/decide-test-datafile.json")
+		datafile, _ = ioutil.ReadFile(absPath)
+	})
+
+	configManager := config.NewStaticProjectConfigManagerWithOptions("", config.WithInitialDatafile(datafile))
+	s.projectConfig, _ = configManager.GetConfig()
+}
+
+func (s *ForcedDecisionServiceTestSuite) TestSetForcedDecision() {
+	s.True(s.forcedDecisionService.SetForcedDecision("1", "2", "3"))
+	s.True(s.forcedDecisionService.SetForcedDecision("1", "2", ""))
+	s.True(s.forcedDecisionService.SetForcedDecision("", "2", "3"))
+	s.True(s.forcedDecisionService.SetForcedDecision("1", "", "3"))
+	s.True(s.forcedDecisionService.SetForcedDecision("1", "", ""))
+	s.True(s.forcedDecisionService.SetForcedDecision("", "2", ""))
+	s.True(s.forcedDecisionService.SetForcedDecision("", "", "3"))
+	s.True(s.forcedDecisionService.SetForcedDecision("", "", ""))
+}
+
+func (s *ForcedDecisionServiceTestSuite) TestGetForcedDecision() {
+	forcedDecision := s.forcedDecisionService.GetForcedDecision("1", "2")
+	s.Equal("", forcedDecision)
+
+	s.True(s.forcedDecisionService.SetForcedDecision("1", "2", "3"))
+	forcedDecision = s.forcedDecisionService.GetForcedDecision("1", "2")
+	s.Equal("3", forcedDecision)
+
+	s.True(s.forcedDecisionService.SetForcedDecision("1", "2", ""))
+	forcedDecision = s.forcedDecisionService.GetForcedDecision("1", "2")
+	s.Equal("", forcedDecision)
+
+	s.True(s.forcedDecisionService.SetForcedDecision("", "2", "3"))
+	forcedDecision = s.forcedDecisionService.GetForcedDecision("", "2")
```

Comment:
Are we going to accept "" flagKey as valid inputs?  It should be consistent with doc later.
If we define valid "flagKey" required, we can check non-empty flagKey when setForcedDecision.

Reply:
Updated `setForcedDecision` to only accept non-empty flagKey.